src/backend/api/meals.js

```diff
@@ -2,14 +2,97 @@ const express = require("express");
 const router = express.Router();
 const knex = require("../database");
 
-router.get("/", async (request, response) => {
+router.get("/title", async (request, response) => {
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
     const titles = await knex("meals").select("title");
     response.json(titles);
   } catch (error) {
     throw error;
   }
 });
 
+//Get
+router.get("/", async (req, res) => {
+  try {
+    const allMeals = await knex.select("*").from("meal");
+    if (allMeals) {
+      res.status(200).json(allMeals);
+    } else {
+      res.status(404).send("Request not found");
+    }
+  } catch (error) {
+    console.error("Server Error", err);
+    res.status(500).json({ error: "Server Error: unable to import data" });
+  }
+});
+
+//Post
+router.post("/", async (req, res) => {
+  try {
+    const newMeal = req.body;
+    const newMealAdded = await knex("meal").insert(newMeal);
+    if (newMealAdded) {
+      res.status(201).json({ message: "good creation" });
+    }
+    res.status(404).send("Request not found");
```

Comment:
I'm not sure if 404 is the proper code in this case (if you are sure it is, please ignore me :) - I'm not an expert in this, but 404 might not be the best choice for post requests since it means not found and we are making a new entry to the DB here